glean-core/src/database/mod.rs

```diff
@@ -104,6 +104,30 @@ fn file_size(path: &Path) -> Option<NonZeroU64> {
         .and_then(NonZeroU64::new)
 }
 
+/// Determine the size of all the files in the given file directory.
+///
+///  # Arguments
+///
+///  -`path` - The path to the directory
+///
+///  # Returns
+///
+/// Returns the non-zero combined size of all files,
+/// or `None` on error or if the size is `0`.
+fn database_size(dir: &Path) -> Option<NonZeroU64> {
+    let mut total_size = 0;
+    if let Ok(entries) = fs::read_dir(dir) {
+        for entry in entries {
+            if let Ok(entry) = entry {
+                let path = entry.path();
+                // the getter is returning the u64 value from NonZeroU64
+                total_size += file_size(&path)?.get();
```

Comment:
We can savae some work here. You can read the [`metadata`](https://doc.rust-lang.org/std/fs/struct.DirEntry.html#method.metadata) of an entry directly, that one then has the [`len`](https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.len). You can then also remove the `file_size` function as its not used anymore.

If there's no metadata for whatever reason, we shouldn't bail out early here. If we can't read the file size of a single file, we should still read the remaining files.